.github/workflows/reusable-tsan.yml

```
@@ -60,3 +64,9 @@ jobs:
       run: make pythoninfo
     - name: Tests
       run: ./python -m test --tsan -j4
+    - name: Archive TSAN logs
+      if: always()
+      uses: actions/upload-artifact@v4
+      with:
+        name: ${{ inputs.tsan_logs_artifact_name }}
+        path: tsan_log.*
```

https://github.com/python/cpython/actions/runs/8993754017?pr=118747 says:

> No files were found with the provided path: tsan_log.*. No artifacts will be uploaded.


Should this match the `tsan_logs_artifact_name` values?

```
      tsan_logs_artifact_name: tsan-logs-default
...
      tsan_logs_artifact_name: tsan-logs-free-threading
```


```suggestion
        path: tsan-log*
```

---

No, I think what's currently in the PR is correct. TSAN will emit its logs into files named `tsan_log.<pid>` ([docs](https://github.com/google/sanitizers/wiki/ThreadSanitizerFlags)). GH actions then packages these into a zip file and stores them as the named artifact.

There are no log files (and hence artifacts) for the run that you linked because TSAN didn't emit any logs. This is expected in the common case (TSAN tests passing, no data races detected). When TSAN fails, the logs will contain the data races that it detects. The [workflow run](https://github.com/mpage/cpython/actions/runs/8993391354) in the PR description is an example of this. I used an incomplete suppressions file for the free-threaded builds so that TSAN will report data races. These races are packaged up into an artifact that is visible from the workflow run.

---

Thanks for the explanation 👍